pkgs/development/tools/godot/default.nix

```diff
@@ -65,39 +83,76 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  sconsFlags = "target=release_debug platform=x11";
-  preConfigure = ''
-    sconsFlags+=" ${
-      lib.concatStringsSep " "
-      (lib.mapAttrsToList (k: v: "${k}=${builtins.toJSON v}") options)
-    }"
-  '';
+  sconsFlags = let toSconsValue = x: if builtins.isBool x then lib.boolToString x else x; in
+    lib.pipe self.passthru.sconsFlagSet [
+      (lib.mapAttrsToList (k: v: "${k}=${toSconsValue v}"))
+      (lib.concatStringsSep " ")
+    ];
 
   outputs = [ "out" "dev" "man" ];
 
+  installSteps = [ "installBinStep" "installDevStep" "installManStep" "installShortcutStep" ];
+
   installPhase = ''
-    mkdir -p "$out/bin"
-    cp bin/godot.* $out/bin/godot
+    runHook preInstall
+
+    for curInstallStep in $installSteps; do
+      eval "''${!curInstallStep}"
+    done
+
+    runHook postInstall
+  '';
+
+  installBinStep = ''
+    runHook preInstallBin
+
+    bin="$out"/"$binInstallPath"
+    mkdir -p "$bin"
+    cp -R bin/. "$bin"
+    mv "$bin"/$builtGodotBinNamePattern "$bin/$installedGodotBinName"
+
+    runHook postInstallBin
+  '';
+
+  installDevStep = ''
+    runHook preInstallDev
 
     mkdir "$dev"
-    cp -r modules/gdnative/include $dev
+    cp -R modules/gdnative/include "$dev"
+
+    runHook postInstallDev
+  '';
+
+  installManStep = ''
+    runHook preInstallMan
+
+    godotman="$man"/share/man/man6
+    mkdir -p "$godotman"
+    cp misc/dist/linux/godot.6 "$godotman"
+
+    runHook postInstallMan
+  '';
+
+  installShortcutStep = ''
+    runHook preInstallShortcut
 
-    mkdir -p "$man/share/man/man6"
-    cp misc/dist/linux/godot.6 "$man/share/man/man6/"
+    share="$out"/share
+    mkdir -p "$share/"{applications,icons/hicolor/scalable/apps}
+    cp misc/dist/linux/org.godotengine.Godot.desktop "$share"/applications
+    cp icon.svg "$share"/icons/hicolor/scalable/apps/godot.svg
+    cp icon.png "$share"/icons/godot.png
+    substituteInPlace "$share"/applications/org.godotengine.Godot.desktop --replace "Exec=godot" "Exec=\"$out/bin/$pname\""
```

Comment:
This is actually not to great because desktops take a bit to update desktop files and this makes them start old versions of programs.

Reply:
This is not a refactor.  I just kept in place what was already there.  I don't know much about desktops in Linux, so I am not in a position to question it.

Reply:
> This is not a refactor.

The PR title suggests otherwise ``godot: refactor and add godot mono``. I am confused now.

Reply:
When I said "this is not a refactor", I was referring to the 4 lines of script this comment is targeting.  Specifically, the steps for copying the .desktop file and icons, and then tweaking the godot path within the shortcut ... are not a refactor.  I copied them from the pre-existing version of the godot derivation.